### PR TITLE
Updated Prom dep; cleaned and fixed leaking unit tests.

### PR DESCRIPTION
**Pull request by bwplotka**
Also fixes: https://github.com/thanos-io/thanos/issues/1442
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/ui/query.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+
```
Comment (povilasv) [inline pkg/ui/query.go]: remove the space

File: pkg/ui/rule.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 	"sort"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+
```
Comment (povilasv) [inline pkg/ui/rule.go]: ditto

File: test/e2e/receive_test.go [hunk 3]
```diff
@@ -12,97 +13,123 @@ import (
...
+
+	a := newLocalAddresser()
+	{
+		// The hashring suite creates three receivers, each with a Prometheus
+		// remote-writing data to it. However, due to the hashing of the labels,
+		// the time series from the Prometheus is forwarded to a different
+		// receiver in the hashring than the one handling the request.
+		// The querier queries all the receivers and the test verifies
+		// the time series are forwarded to the correct receive node.
+
```
Comment (squat) [inline test/e2e/receive_test.go]: nit: extra newline here

File: test/e2e/receive_test.go [hunk 4]
```diff
@@ -12,97 +13,123 @@ import (
...
+		receiveHTTP1, receiveHTTP2, receiveHTTP3 := a.New(), a.New(), a.New()
+
+		r1 := receiver(receiveHTTP1, a.New(), a.New(), 1, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r2 := receiver(receiveHTTP2, a.New(), a.New(), 1, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r3 := receiver(receiveHTTP3, a.New(), a.New(), 1, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+
+		prom1 := prometheus(a.New(), defaultPromRemoteWriteConfig(nodeExporterHTTP(1), remoteWriteEndpoint(r1.HTTP)))
```
Comment (squat) [inline test/e2e/receive_test.go]: Do we need to still have both `addRemoteWriteToConfig(defaultPromConfig(...))` as well as `defaultPromRemoteWriteConfig`?

File: test/e2e/receive_test.go [hunk 5]
```diff
@@ -12,97 +13,123 @@ import (
...
+		// receivers and the test verifies that the time series are
+		// replicated to all of the nodes.
+		receiveHTTP1, receiveHTTP2, receiveHTTP3 := a.New(), a.New(), a.New()
+
+		r1 := receiver(receiveHTTP1, a.New(), a.New(), 3, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r2 := receiver(receiveHTTP2, a.New(), a.New(), 3, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r3 := receiver(receiveHTTP3, a.New(), a.New(), 3, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+
+		prom1 := prometheus(a.New(), defaultPromRemoteWriteConfig(nodeExporterHTTP(1), remoteWriteEndpoint(r1.HTTP)))
+		prom2 := prometheus(a.New(), defaultPromConfig("no-remote-write", 2, prom1.HTTP))
```
Comment (squat) [inline test/e2e/receive_test.go]: previously, every receive component also created a prometheus, that is why we had two proms with `no-remote-write`. Now that they are independent, we can eliminate `prom2` and `prom3` from here. They are not doing anything

File: test/e2e/receive_test.go (hunk 4 above)
Comment (bwplotka) [reply test/e2e/receive_test.go]: Nope, removed.

File: test/e2e/receive_test.go (hunk 5 above)
Comment (bwplotka) [reply test/e2e/receive_test.go]: :man_facepalming:

**Comment by bwplotka** [comment]
Finally, tests fixed, all green again!

**Pull request merged**